Make off canvas work no matter where you are on the page...

brettsmason opened the pull request:
... and avoid having to jump to the top every time. This also now works when using a sticky menu etc.
See [this CodePen](http://codepen.io/brettsmason/pen/WxeeNO) for an example of how it works.

This could use some cross device/browser testing to confirm it works correctly.

Closes #8836

brettsmason committed:
Make off canvas work no matter where you are on the page and avoids having to jump to the top.

Silvester20 commented:
Altough https://github.com/zurb/foundation-sites/issues/8050 was closed, the 'scroll to top' issue was never fixed. But maybe this PR can fix it finally. You are awesome @brettsmason

rafibomb commented:
@Silvester20 How did this work for you? The more people that report the PR fixes the issue the quicker we can merge it :) And @brettsmason is awesome!

Silvester20 commented:
@rafibomb I am not able to test this for my own site but looking at the codepen provided by Brett, it seems to work.

rafibomb commented:
@brettsmason Do you know why the travis failed?

brettsmason commented:
@rafibomb I have no idea. If anyone has a clue let me know and ill look into correcting it.

brettsmason commented:
@rafibomb Not knowing anything about travis it looks like the `npm install` command failed. Is there any way to restart the checks?

kball commented:
Restarted and they appear to have passed this time.

deepfriedmind commented:
@brettsmason: In your [pen](http://codepen.io/brettsmason/pen/WxeeNO) you have the following SCSS:

```scss
.off-canvas-content {
  overflow-y: auto;
}
```

But this but seems to be missing from your [pull request](https://github.com/zurb/foundation-sites/pull/8872/files), which would make the container unable to scroll. Or am I missing something?

rafibomb commented:
Really want to get this in - amazing PR @brettsmason. There is just one outstanding question on it by @deepfriedmind can you take a look?

Also should be pointed at branch `v6.3` as it's a pretty amazing feature!

brettsmason commented:
@deepfriedmind @rafibomb Sorry not had time to get back to you on this until now. I'll take a look this evening, but yep I think you are right, I need to add that in (must of missed it by mistake). 

The only thing I'm not completely happy with is the visible scrollbar on the content when you open a right off canvas menu (see the Codepen) but I cant think of a way to solve that other than set content to `overflow: hidden` but that makes the content change width slightly.

I'd love an opinion before I go ahead and fix that.

brettsmason committed:
Add missing overflow style

Add in missing `overflow-y: auto` to `.off-canvas-content` to allow for vertical scrolling.

wonkeythemonkey commented:
One thing I noticed about this solution is that it affects scripting events that are relative to `window` because all scrolling now happens within a div without scrolling the entire viewport. For example, when I applied the four stylesheet rules to the various off-canvas components, it broke one of my scripts that detects scrolling. I had to move a listener that was assigned to `$(window).scroll()` to  `$('.off-canvas-content').scroll()`. It works in my case, but it may have unintended consequences and cause problems when that cannot be easily changed, like in a third-party library.

kball commented:
Unfortunately I'm going to have to revert this... I hadn't fully grokked the impact of @wonkeythemonkey's note, but it caused a slew of unexpected consequences throughout other elements of the framework such as sticky and magellan
